src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -0,0 +1,219 @@
+import org.prebid.server.bidder.consumable.model.ConsumablePlacement;
+import org.prebid.server.bidder.consumable.model.ConsumablePricing;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.consumable.ExtImpConsumable;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class ConsumableBidder implements Bidder<ConsumableBidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpConsumable>> CONSUMABLE_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpConsumable>>() {
+            };
+
+    private final String endpointUrl;
+
+    public ConsumableBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
+
+    @Override
+    public Result<List<HttpRequest<ConsumableBidRequest>>> makeHttpRequests(BidRequest request) {
+        final ConsumableBidRequest.ConsumableBidRequestBuilder requestBuilder = ConsumableBidRequest.builder()
+                .time(Instant.now().getEpochSecond())
+                .includePricingData(true)
+                .enableBotFiltering(true)
+                .parallel(true);
+
+        final Site site = request.getSite();
+        if (site != null) {
+            requestBuilder
+                    .referrer(site.getRef())
+                    .url(site.getPage());
+        }
+
+        try {
+            resolveRequestFields(requestBuilder, request.getImp());
+        } catch (PreBidException e) {
+            return Result.of(Collections.emptyList(), Collections.singletonList(BidderError.badInput(e.getMessage())));
+        }
+
+        final ConsumableBidRequest outgoingRequest = requestBuilder.build();
+        String body;
+        try {
+            body = Json.encode(outgoingRequest);
+        } catch (EncodeException e) {
+            return Result.of(Collections.emptyList(),
+                    Collections.singletonList(BidderError.badInput(
+                            String.format("Failed to encode request body, error: %s", e.getMessage()))));
+        }
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<ConsumableBidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .body(body)
+                        .headers(resolveHeaders(request))
```

Comment:
The `resolveHeaders(...)` method can throw exception, pls move it to `try-catch`.

Reply:
The way they did it in Go is that if there's any error during url validation - they just don't add the header. I'll add this logic to method body.